src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitForeachTests.cs

```diff
@@ -6082,7 +6844,13 @@ internal static class Extensions
 {
     public static C.Enumerator GetAsyncEnumerator(this C self) => new C.Enumerator();
 }";
-            CompileAndVerify(source, parseOptions: TestOptions.RegularPreview, expectedOutput: "123");
+            var comp = CreateCompilationWithMscorlib46(source, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (8,33): warning CS0279: 'C' does not implement the 'async streams' pattern. 'C.GetAsyncEnumerator()' is not a public instance or extension method.
```

Comment:
This warning seems spurious.

Reply:
The warning is to explain why a promising looking method doesn't implement the GetEnumerator pattern. We tried to remove the warning in some cases, but in the end you felt the complexity and arbitrariness wasn't worth it, and asked me to revert the changes.

Reply:
Yeah... Let's log a diagnostic clarity bug for this. It's definitely tricky to fix, but it's also definitely one of those warnings that isn't helpful.

Reply:
Alright, after thinking about this a bit more, I think the warning is fine. I mean, who's going to make an internal `GetEnumerator()` method that otherwise fulfills the pattern? (no doubt someone will point at this comment in a year and laugh)